submarine-commons/commons-unixusersync/src/test/java/org/apache/submarine/commons/unixusersync/EmbeddedLdapRuleTest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.submarine.commons.unixusersync;
+
+import com.google.common.collect.Iterators;
+
+import com.unboundid.ldap.sdk.AddRequest;
+import com.unboundid.ldap.sdk.Attribute;
+import com.unboundid.ldap.sdk.LDAPConnection;
+import com.unboundid.ldap.sdk.LDAPInterface;
+import com.unboundid.ldap.sdk.SearchRequest;
+import com.unboundid.ldap.sdk.SearchResult;
+import com.unboundid.ldap.sdk.SearchResultEntry;
+import com.unboundid.ldap.sdk.SearchScope;
+import org.junit.Rule;
+import org.junit.Test;
+import org.zapodot.junit.ldap.EmbeddedLdapRule;
+import org.zapodot.junit.ldap.EmbeddedLdapRuleBuilder;
+
+import javax.naming.Context;
+import javax.naming.NameClassPair;
+import javax.naming.NamingEnumeration;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.SearchControls;
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+
+public class EmbeddedLdapRuleTest {
+
+  public static final String DOMAIN_DSN = "dc=zapodot,dc=org";
+  @Rule
+  public EmbeddedLdapRule embeddedLdapRule = EmbeddedLdapRuleBuilder
+          .newInstance()
+          .usingDomainDsn(DOMAIN_DSN)
+          .importingLdifs("example.ldif")
+          .build();
+
+  @Test
+  public void testLdapConnection() throws Exception {
+    final LDAPInterface ldapConnection = embeddedLdapRule.ldapConnection();
+    final SearchResult searchResult =
+            ldapConnection.search(DOMAIN_DSN, SearchScope.SUB, "(objectClass=person)");
+
+    assertEquals(2, searchResult.getEntryCount());
+  }
+
+  @Test
+  public void testRawLdapConnection() throws Exception {
+    final String commonName = "Test person";
+    final String dn = String.format(
+            "cn=%s,ou=people,dc=zapodot,dc=org", commonName);
+    LDAPConnection ldapConnection = embeddedLdapRule.unsharedLdapConnection();
+    try {
+      ldapConnection.add(new AddRequest(dn, Arrays.asList(
+              new Attribute("objectclass", "top",
+                      "person", "organizationalPerson", "inetOrgPerson"),
```

Comment:
No line breaks

Reply:
Well... there will more than 110 words without this line breaks.